src/cargo/util/toml.rs

```diff
@@ -737,7 +737,12 @@ impl TomlManifest {
                -> CargoResult<Vec<(PackageIdSpec, Dependency)>> {
         let mut replace = Vec::new();
         for (spec, replacement) in self.replace.iter().flat_map(|x| x) {
-            let spec = try!(PackageIdSpec::parse(spec));
+            let spec = match PackageIdSpec::parse(spec) {
+                Ok(spec) => spec,
+                Err(_)   => bail!("replacements must specify a \
+                                   valid semver version \
+                                   to replace, but `{}` does not", spec),
+            };
```

Comment:
I tried that but It was just displaying:

```
Caused by:
  Parser Error
```

And I couldn't find that very helpful. But I can use `chain_error` if you think is better.